docs/cross-platform.md

```diff
@@ -0,0 +1,44 @@
+# Cross-Platform Development with `esy`
+
+## Goal
+
+The goal for `esy` is to work equally well on Linux, Windows, and OSX - without needing special cases or hacks.
+
+The development workflow for `esy` should be the same for all of these platforms.
+
+## Architecture
+
+One challenge with the OCaml ecosystem is that many of the scripts rely heavily on 'bashisms', like `make`, bash shell scripts, etc.
+
+Unfortunately, those tools aren't available out-of-the-box on Windows. WSL is starting to offer more capabilities around this,
+but there isn't a seamless install story. An important piece of meeting our goal stated above is that the installation process
+for `esy` is simple - `npm install -g esy`.
+
+To that end, we use `cygwin` on Windows to provide a transparent layer for running these 'bashisms'. You don't even have to know about it -
```

Comment:
Cygwin and mingw are some of the most misunderstood technologies. I think it would be nice to reenforce here that `esy` itself will be a full windows native binary, and that  for the most part (except for `rsync`/`git`), cygwin is only there because opam packages often have hard coded assumptions on unix paths/commands etc.